acme/acme/dns_resolver_test.py

```diff
@@ -21,33 +25,56 @@ def create_txt_response(name, txt_records):
     return dns.rrset.from_text_list(name, 60, "IN", "TXT", txt_records)
 
 
-class TxtRecordsForNameTest(unittest.TestCase):
+@test_util.skip_unless(DNS_AVAILABLE,
+                       "optional dependency dnspython is not available")
+class DnsResolverTestWithDns(unittest.TestCase):
+    """Tests for acme.dns_resolver when dns is available."""
+    @classmethod
+    def _call(cls, name):
+        from acme import dns_resolver
```

Comment:
This `_call` syntax violates that scope rule though. We're using it because every function in here needs `from acme import dns_resolver`, which means the import should probably be at the class level anyway. If you want to shorten the name of the function we're testing, that's fine to leave like this, but re-importing for every function seems weird.